### PR TITLE
Fix async desugaring providing wrong input to procedural macros.

### LINKED ISSUE #60674: Async fn loses mut pattern when passed to attribute macro
#### Cargo.toml

```toml
[package]
name = "repro"
version = "0.0.0"
edition = "2018"

[lib]
proc-macro = true
```

#### src/lib.rs

```rust
extern crate proc_macro;
use proc_macro::TokenStream;

#[proc_macro_attribute]
pub fn attr(_args: TokenStream, input: TokenStream) -> TokenStream {
    println!("{}", input);
    TokenStream::new()
}
```

#### src/main.rs

```rust
#[repro::attr]
async fn f(mut x: u8) {}

fn main() {}
```

Between nightly-2019-05-07 and nightly-2019-05-08, the `mut` pattern that should be part of the macro input went missing. https://github.com/rust-lang/rust/compare/a19cf18c7dbbcc46dddea81df3a4cee1735c2349...cfdc84a009020c59e53e4039beae22eb59e41685

```console
$ cargo +nightly-2019-05-07 check
   Compiling repro v0.0.0
async fn f(mut x: u8) { }

$ cargo +nightly-2019-05-08 check
   Compiling repro v0.0.0
async fn f(x: u8) { }
```

Originally reported by @kjvalencik in https://github.com/dtolnay/syn/issues/632.
Mentioning @taiki-e @cramertj @davidtwco because #60535 looks relevant and is in the right commit range.

### PR DESCRIPTION
**Pull request by davidtwco**
Fixes #60674. 

This PR fixes a minor oversight introduced by #60535 where unused `mut` binding modes were removed from the arguments to an `async fn` (as they were added to the statement that we insert into the closure body). However, this meant that the input to procedural macros was incorrect. This removes that and instead fixes the `unused_mut` error that it avoided.

r? @cramertj 
cc @taiki-e

### CODE REVIEW

File: src/test/ui/async-await/issue-60674.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// edition:2018
+#![feature(async_await)]
+
+// This is a regression test that ensures that `mut` patterns are not lost when provided as input
+// to a proc macro.
+
+extern crate issue_60674;
+
+#[issue_60674::attr]
+async fn f(mut x: u8) {}
```
Comment (Centril) [inline src/test/ui/async-await/issue-60674.rs]: @davidtwco Can you follow up with a PR that extends this test to `mut` patterns in nested places?

Comment (davidtwco) [reply src/test/ui/async-await/issue-60674.rs]: Sure.

Comment (davidtwco) [reply src/test/ui/async-await/issue-60674.rs]: Submitted #60692.

**Pull request merged**